src/harvest.h

```diff
@@ -23,6 +23,8 @@ struct harvest_entry {
     std::pair<float, float> scale_num = { 0.0f, 0.0f };
 
     int max = 1000;
+    itype_id type = "null";
```

Comment:
What's the difference between this `type` and the existing `drop`?

It appears you only use this member to compare it to fixed strings, so it's not really used as item type id. An enumeration seems a better fit for this. At least don't make it a `itype_id` because that's what it clearly is *not* used for.